File: embedding.py
from transformers import AutoTokenizer, AutoModel
import torch
import sys
import json

model_name = "sentence-transformers/all-MiniLM-L6-v2"
tokenizer = AutoTokenizer.from_pretrained(model_name)
model = AutoModel.from_pretrained(model_name)

def generate_embedding(text):
    inputs = tokenizer(text, return_tensors="pt", truncation=True, padding=True)
    with torch.no_grad():
        embeddings = model(**inputs).last_hidden_state.mean(dim=1)
    return embeddings[0].numpy().tolist()

if __name__ == "__main__":
    text = sys.stdin.read()
    embedding = generate_embedding(text)
    print(json.dumps(embedding))

File: embedding.js
import AutoTokenizer from "transformers"
import AutoModel from "transformers"
import torch from "torch"
const modelName = "sentence-transformers/all-MiniLM-L6-v2";

import {spawn} from "child_process"
async function generateEmbedding(text) {
  return new Promise((resolve, reject) => {
    const pythonProcess = spawn("python", ["./embedding.py"]);


    pythonProcess.stdin.write(text);
    pythonProcess.stdin.end();

    let data = "";

    pythonProcess.stdout.on("data", (chunk) => {
      data += chunk;
    });

    pythonProcess.stderr.on("data", (error) => {
        console.error("Ошибка Python скрипта (stderr):", error.toString());
        reject(error.toString());
      });
      
      pythonProcess.stdout.on("data", (data) => {
        console.log("Вывод Python скрипта (stdout):", data.toString());
      });
      

    pythonProcess.on("close", (code) => {
      if (code === 0) {
        resolve(JSON.parse(data));
      } else {
        reject(`Python скрипт завершился с кодом ${code}`);
      }
    });
  });
}


export default generateEmbedding;
